File: Basic/Game with String.cpp
//{ Driver Code Starts
//Initial Template for C++

#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
//User function Template for C++

class Solution
{
public:
    int minValue(string s, int k)
    {
        // code here
        int n = s.length();
        int mappa[27] = {};
        
        for (int i=0; i<n; i++)
        {
            mappa[s[i] - 'a']++;
        }
        
        while (k)
        {
            sort(mappa, mappa+27);
            mappa[26]--;
            k--;
        }
        
        int ans = 0;
        
        for (int i=0; i<27; i++)
        {
            ans += mappa[i] * mappa[i];
        }
        
        return ans;
        
    }
};

//{ Driver Code Starts.

int main(){
    int t;
    cin>>t;
    while(t--){
        string s;
        int k;
        cin>>s>>k;
        
        Solution ob;
        cout<<ob.minValue(s, k)<<"\n";
    }
    return 0;
}
// } Driver Code Ends

File: Medium/Game with String.py
#User function Template for python3
import heapq
from collections import Counter


class Solution:
    def minValue(self, s, k):
        # code here
        counter_obj = Counter(s)
        
        tarr = list(counter_obj.items())
        tarr = sorted(tarr, key=lambda x: x[1])
        
        arr = [list(it[::-1]) for it in tarr]
        
        h = []
        
        for it in arr:
            it[0] = -1 * it[0] 
            heapq.heappush(h, it)
            
        while k > 0:
            temp = heapq.heappop(h)
            temp[0] +=1 
            heapq.heappush(h, temp)
            k -= 1
        
        ans = 0
        
        for it in h:
            ans += it[0] ** 2
        
        return ans
        
        
        

#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    t = int(input())
    for _ in range(t):
        s = input()
        k = int(input())
        
        ob = Solution()
        print(ob.minValue(s, k))
# } Driver Code Ends
